pkg/metrics/server_test.go

```
@@ -0,0 +1,67 @@
+	collector := fixedCollector{desc: prometheus.NewDesc("cursus_test_runtime_value", "test", nil, nil)}
+	server, err := StartMetricsServerAddress("127.0.0.1:0", collector)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+		defer cancel()
+		_ = server.Shutdown(ctx)
+	})
+
+	response, err := http.Get("http://" + server.Addr + "/metrics") // #nosec G107 -- test server address is loopback-only.
```

golangci-lint (2.12.2)

[error] net/http.Get must not be called. use net/http.NewRequestWithContext and (*net/http.Client).Do(*http.Request)

(noctx)